src/UploadClient.ts

```diff
@@ -4,72 +4,47 @@ import fileFrom from './fileFrom/fileFrom'
 import groupFrom from './groupFrom/groupFrom'
 
 /* Types */
-import {FileFromEnum} from './fileFrom/types'
+import {FileFromEnum, FileUploadInterface} from './fileFrom/types'
 import {FileData, SettingsInterface} from './types'
-import {FileUploadInterface} from './fileFrom/types'
 import {Url} from './api/fromUrl'
 import {UploadAPIInterface, Uuid} from './api/types'
 import {GroupFromEnum, GroupUploadInterface} from './groupFrom/types'
 
 export interface UploadClientInterface {
-  setSettings(newSettings: SettingsInterface): void
+  updateSettings(newSettings: SettingsInterface): void
 
   getSettings(): SettingsInterface
 
-  addUpdateSettingsListener(listener: Function): void
-
-  removeUpdateSettingsListener(listener: Function): void
-
   fileFrom(from: FileFromEnum, data: FileData | Url | Uuid, settings?: SettingsInterface): FileUploadInterface
 
   groupFrom(from: GroupFromEnum, data: FileData[] | Url[] | Uuid[], settings?: SettingsInterface): GroupUploadInterface
 }
 
 class UploadClient implements UploadClientInterface {
-  protected settings: SettingsInterface
-  protected updateSettingsListeners: Function[]
+  private settings: SettingsInterface
   readonly api: UploadAPIInterface
 
   constructor(settings: SettingsInterface = {}) {
     this.settings = {
       ...defaultSettings,
       ...settings,
     }
-    this.updateSettingsListeners = []
     this.api = new UploadAPI(this)
   }
 
-  setSettings(newSettings: SettingsInterface = {}): void {
+  updateSettings(newSettings: SettingsInterface = {}): void {
     const prevSettings = {...this.settings}
 
     this.settings = {
       ...prevSettings,
```

Comment:
why we copy `this.settings` twice?

Reply:
@jeetiss We just merge with previous set settings. Do you think that we need overwrite?

Reply:
I think code in 36 line useless and need rewrite method and remove `prevSettings` var